### DISCUSSION TITLE
Hello, Getting EPERM Error when trying to save to External Disk

### DISCUSSION THREAD

**Discussion by amfmxm** [aclap-dev/video-downloadhelper#91] [Help and Q&A]
### Your question

Error: EPERM: operation not permitted, mkdir 'E:\'
    at rs.receive (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:13157)
    at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42562

When trying to save to external drive ( E:\ ) , using Chrome extension 8.2.0.23 version Feb 14 2024 update verfied edition. Saving to local disk works however. Thanks

### Addon & CoApp version and details

Video DownloadHelper
Version 8.2.0.23 (stable)
Browser locale: en-US
Production build
Built on 2024-02-14
Build options: browser=chrome, linuxlic=false, noyt=true, target=google

Platform Win x86-32
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36

Found companion app: VdhCoApp 2.0.17

License verified

**Comment by paulrouget** [comment]
Oh right, I heard of that one before ([here](https://github.com/aclap-dev/vdhcoapp/issues/199)). I'll investigate.

**Reply by amfmxm** [reply]
Thank you very much for this and also the prompt updates and fixes these past few days

I should add that my differences between mine and that users are that my video in question is a DASH stream video where the website requires login to acccess. Also, that user mentions Twitch clips vs Youtube clips. Im not familiar if Twitch clips are also DASH streams but Youtube videos I use a separate app for.

**Comment by paulrouget** [comment]
Can you try this and tell me if this fixes the EPerm error:

- right click on the parent folder of the directory you want to download in
- Properties > Security Tab
- Click on Edit to change Permissions
- Select the user and give Full Control 

?

**Reply by amfmxm** [reply]
No fix, but I did just learn that it will save to folder within external drive if that helps out in any way

**Reply by paulrouget** [reply]
Yeah, EPerm error happens when saving video on a different hard drive. The CoApp doesn't have the right to write things there. I'm surprised the above-mentionned fix doesn't work.

Do the download directory already exist?

**Reply by amfmxm** [reply]
Yes even when setting download directory still get error

**Comment by paulrouget** [comment]
I think I figured out a way around this issue. Give me a couple of days…

**Comment by paulrouget** [comment]
We will publish version 8.2.0.24 in the coming days, which, I  hope, will fix the EPerm situation.

**Reply by amfmxm** [reply]
Thanks for the speedy communication 👍 Sounds great looking forward to it
